fix: フロントエンドのCIがlintのeslint-plugin-react-hooksで落ちる

cf. https://github.com/poporonnet/kcmsx/actions/runs/9887445740/job/27309164191

```
TypeError: context.getSource is not a function
Occurred while linting /home/runner/work/kcmsx/kcmsx/packages/kcmsf/src/components/match/PointCountable.tsx:21
Rule: "react-hooks/exhaustive-deps"
    at visitFunctionWithDependencies (/home/runner/work/kcmsx/kcmsx/node_modules/eslint-plugin-react-hooks/cjs/eslint-plugin-react-hooks.development.js:1559:256)
    at visitCallExpression (/home/runner/work/kcmsx/kcmsx/node_modules/eslint-plugin-react-hooks/cjs/eslint-plugin-react-hooks.development.js:1759:[11](https://github.com/poporonnet/kcmsx/actions/runs/9887445740/job/27309164191#step:5:12))
    at ruleErrorHandler (/home/runner/work/kcmsx/kcmsx/node_modules/eslint/lib/linter/linter.js:1113:48)
    at /home/runner/work/kcmsx/kcmsx/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/home/runner/work/kcmsx/kcmsx/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/home/runner/work/kcmsx/kcmsx/node_modules/eslint/lib/linter/node-event-generator.js:297:26)
    at NodeEventGenerator.applySelectors (/home/runner/work/kcmsx/kcmsx/node_modules/eslint/lib/linter/node-event-generator.js:326:22)
    at NodeEventGenerator.enterNode (/home/runner/work/kcmsx/kcmsx/node_modules/eslint/lib/linter/node-event-generator.js:340:[14](https://github.com/poporonnet/kcmsx/actions/runs/9887445740/job/27309164191#step:5:15))
    at runRules (/home/runner/work/kcmsx/kcmsx/node_modules/eslint/lib/linter/linter.js:1157:40)
Error: Process completed with exit code 2.
```

eslint-plugin-react-hooksがeslint v9に対応していない模様 @eslint/compatを通す必要がある？

--- a/packages/kcmsf/eslint.config.js
+++ b/packages/kcmsf/eslint.config.js
@@ -1,14 +1,17 @@
+import { fixupPluginRules } from "@eslint/compat";
 import { FlatCompat } from "@eslint/eslintrc";
 import js from "@eslint/js";
 import typeScriptESLint from "@typescript-eslint/eslint-plugin";
 import typeScriptESLintParser from "@typescript-eslint/parser";
 import eslintConfigPrettier from "eslint-config-prettier";
+import reactHooks from "eslint-plugin-react-hooks";
 import reactRefresh from "eslint-plugin-react-refresh";
 import vitest from "eslint-plugin-vitest";
 import globals from "globals";
 
 const compat = new FlatCompat();
 
+/** @type {import("eslint").Linter.FlatConfig} */
 export default [
   {
     files: ["src/**/*.{ts,tsx}"],
@@ -19,7 +22,6 @@ export default [
   js.configs.recommended,
   eslintConfigPrettier,
   ...compat.extends("plugin:@typescript-eslint/recommended"),
-  ...compat.extends("plugin:react-hooks/recommended"),
   {
     languageOptions: {
       parser: typeScriptESLintParser,
@@ -32,12 +34,14 @@ export default [
     plugins: {
       "@typescript-eslint": typeScriptESLint,
       "react-refresh": reactRefresh,
+      "react-hooks": fixupPluginRules(reactHooks),
       vitest,
     },
   },
   {
     rules: {
       ...vitest.configs.recommended.rules,
+      ...reactHooks.configs.recommended.rules,
       "react-refresh/only-export-components": [
         "warn",
         { allowConstantExport: true },

--- a/packages/kcmsf/package.json
+++ b/packages/kcmsf/package.json
@@ -26,6 +26,7 @@
     "react-timer-hook": "^3.0.7"
   },
   "devDependencies": {
+    "@eslint/compat": "^1.1.1",
     "@testing-library/jest-dom": "^6.4.2",
     "@testing-library/react": "^16.0.0",
     "@testing-library/user-event": "^14.5.2",

